src/Features/CSharp/Portable/Completion/KeywordRecommenders/MethodKeywordRecommender.cs

```diff
@@ -27,13 +26,13 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
             var token = context.TargetToken;
 
             if (token.Kind() == SyntaxKind.OpenBracketToken &&
-                token.Parent.IsKind(SyntaxKind.AttributeList))
+                token.Parent.IsKind(SyntaxKind.AttributeList) &&
+                token.Parent.FirstAncestorOrSelf<SyntaxNode>(n => n is
+                    PropertyDeclarationSyntax or
+                    EventDeclarationSyntax or
+                    ClassDeclarationSyntax { ParameterList: not null }) is not null)
```

Comment:
I assume we want to handle records here as well.

Reply:
_I_ most certainly do not 😛